source/common/network/win32_socket_handle_impl.h

```diff
@@ -20,7 +21,8 @@ class Win32SocketHandleImpl : public IoSocketHandleImpl {
 public:
   explicit Win32SocketHandleImpl(os_fd_t fd = INVALID_SOCKET, bool socket_v6only = false,
                                  absl::optional<int> domain = absl::nullopt)
-      : IoSocketHandleImpl(fd, socket_v6only, domain) {}
+      : IoSocketHandleImpl(fd, socket_v6only, domain),
+        peek_buffer_(std::make_unique<Buffer::OwnedImpl>()) {}
```

Comment:
Should it be lazily allocated on peek and de-allocated when peek buffer becomes fully drained?